[#6952] fix odd app behavior after quick logout

rasom opened the pull request:
fix #6952 

## Node instance management ##

Currently the node instance with "dummy" configs is used to allow 
`CreateAccount` and `RecoverAccount` calls, but there is no point in doing this 
as node can be started right before these calls. Also we restart node on logging
out which causes weird behaviour like described here #6952.

**How it worked in `develop` before this PR**:
1. `node` is started on application startup.
   `PFSEnabled=false,NoDiscovery=true` are used, so it is started only to make
   possible `CreateAccount` and `RecoverAccount` calls, because they require
   running node atm.
2. When the user **presses "Sign in" button**, the node is stopped, then started
   with user specific configs (`InstallationID`, custom bootnodes, etc), and
   only after that `Login` call is performed.
3. When the user **creates** a new account, at first `CreateAccount` call happens,
   and after that node is stopped and started as in **1.**.
4. When the user **restores** their account, at first `RecoverAccount` call happens,
   and after that node is stopped and started as in **1.**.
5. When the user **logs out** the node is stopped and started again as in **1.** to
   make possible CreateAccount`,`RecoverAccount` calls.

**How it works in this PR**:
1. `node` is not started on the app startup
2. When the user **presses "Sign in" button** the node is started
   with user specific configs (`InstallationID`, custom bootnodes, etc), and
   only after that `Login` call is performed.
3. When the user **creates** a new account, at first the node is started with
   default params (the same as would be used when user signs into the app after
   account creation whithout changing any setting), then `CreateAccount` call
   happens, then `Login`.
4. When the user **restores** their account, the flow is the same as **3.** but
   with `RecoverAccount` instead of `CreateAccount`
5. When the user **logs out** the node is stopped. That's it.

status: ready <!-- Can be ready or wip -->

ghost commented:
<!--prchecklist--> 
### Pull Request Checklist
- [ ] Have you updated the documentation, if impacted (e.g. [docs.status.im](https://docs.status.im/docs/build_status.html))?

rasom committed:
[#6952] Fix odd app behavior after quick logout

This commit changes the way how/when the node is started/stopped:
1. `node` is not started on the app startup
2. When the user presses "Sign in" button the node is started
   with user specific configs (`InstallationID`, custom bootnodes, etc),
   and only after that `Login` call is performed.
3. When the user creates a new account, at first the node is started
   with default params (the same as would be used when user signs into
   the app after account creation whithout changing any setting), then
   `CreateAccount` call happens, then `Login`.
4. When the user restores their account, the flow is the same as `3`
   but with `RecoverAccount` instead of `CreateAccount`
5. When the user logs out the node is stopped. That's it.

yenda reviewed (approved):
Thanks!

status-im-auto commented:
#### :x: [status-react/prs/macos/PR-7032#2](https://ci.status.im/job/status-react/job/prs/job/macos/job/PR-7032/2/) :small_orange_diamond: ~6 min 30 sec :small_orange_diamond: 55f78f3a :small_orange_diamond: [:page_facing_up: build log](https://ci.status.im/job/status-react/job/prs/job/macos/job/PR-7032/2//consoleText)

status-im-auto commented:
#### :x: [status-react/prs/android/PR-7032#2](https://ci.status.im/job/status-react/job/prs/job/android/job/PR-7032/2/) :small_orange_diamond: ~6 min 42 sec :small_orange_diamond: 55f78f3a :small_orange_diamond: [:page_facing_up: build log](https://ci.status.im/job/status-react/job/prs/job/android/job/PR-7032/2//consoleText)

status-im-auto commented:
#### :x: [status-react/prs/ios-e2e/PR-7032#2](https://ci.status.im/job/status-react/job/prs/job/ios-e2e/job/PR-7032/2/) :small_orange_diamond: ~6 min 46 sec :small_orange_diamond: 55f78f3a :small_orange_diamond: [:page_facing_up: build log](https://ci.status.im/job/status-react/job/prs/job/ios-e2e/job/PR-7032/2//consoleText)

status-im-auto commented:
#### :x: [status-react/prs/android-e2e/PR-7032#2](https://ci.status.im/job/status-react/job/prs/job/android-e2e/job/PR-7032/2/) :small_orange_diamond: ~7 min 11 sec :small_orange_diamond: 55f78f3a :small_orange_diamond: [:page_facing_up: build log](https://ci.status.im/job/status-react/job/prs/job/android-e2e/job/PR-7032/2//consoleText)

status-im-auto commented:
#### :x: [status-react/prs/ios/PR-7032#2](https://ci.status.im/job/status-react/job/prs/job/ios/job/PR-7032/2/) :small_orange_diamond: ~7 min 29 sec :small_orange_diamond: 55f78f3a :small_orange_diamond: [:page_facing_up: build log](https://ci.status.im/job/status-react/job/prs/job/ios/job/PR-7032/2//consoleText)

status-im-auto commented:
#### :x: [status-react/prs/linux/PR-7032#2](https://ci.status.im/job/status-react/job/prs/job/linux/job/PR-7032/2/) :small_orange_diamond: ~8 min 5 sec :small_orange_diamond: 55f78f3a :small_orange_diamond: [:page_facing_up: build log](https://ci.status.im/job/status-react/job/prs/job/linux/job/PR-7032/2//consoleText)

status-im-auto commented:
#### :x: [status-react/prs/windows/PR-7032#2](https://ci.status.im/job/status-react/job/prs/job/windows/job/PR-7032/2/) :small_orange_diamond: ~10 min :small_orange_diamond: 55f78f3a :small_orange_diamond: [:page_facing_up: build log](https://ci.status.im/job/status-react/job/prs/job/windows/job/PR-7032/2//consoleText)

status-im-auto commented:
#### :x: [status-react/prs/ios/PR-7032#3](https://ci.status.im/job/status-react/job/prs/job/ios/job/PR-7032/3/) :small_orange_diamond: ~54 sec :small_orange_diamond: a71234a4 :small_orange_diamond: [:page_facing_up: build log](https://ci.status.im/job/status-react/job/prs/job/ios/job/PR-7032/3//consoleText)

status-im-auto commented:
#### :heavy_check_mark: [status-react/prs/macos/PR-7032#3](https://ci.status.im/job/status-react/job/prs/job/macos/job/PR-7032/3/) :small_blue_diamond: ~12 min :small_blue_diamond: a71234a4 :small_blue_diamond: [:package: macos package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-181212-181034-a71234-pr.dmg)

status-im-auto commented:
#### :heavy_check_mark: [status-react/prs/linux/PR-7032#3](https://ci.status.im/job/status-react/job/prs/job/linux/job/PR-7032/3/) :small_blue_diamond: ~21 min :small_blue_diamond: a71234a4 :small_blue_diamond: [:package: linux package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-181212-181033-a71234-pr.AppImage)

status-im-auto commented:
#### :heavy_check_mark: [status-react/prs/macos/PR-7032#4](https://ci.status.im/job/status-react/job/prs/job/macos/job/PR-7032/4/) :small_blue_diamond: ~13 min :small_blue_diamond: a46ed2cc :small_blue_diamond: [:package: macos package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-181213-082308-a46ed2-pr.dmg)

status-im-auto commented:
#### :heavy_check_mark: [status-react/prs/android-e2e/PR-7032#4](https://ci.status.im/job/status-react/job/prs/job/android-e2e/job/PR-7032/4/) :small_blue_diamond: ~14 min :small_blue_diamond: a46ed2cc :small_blue_diamond: [:package: android e2e package](https://status-im.ams3.digitaloceanspaces.com/StatusIm-181213-082302-a46ed2-e2e.apk)

status-im-auto commented:
#### :heavy_check_mark: [status-react/prs/ios-e2e/PR-7032#4](https://ci.status.im/job/status-react/job/prs/job/ios-e2e/job/PR-7032/4/) :small_blue_diamond: ~14 min :small_blue_diamond: a46ed2cc :small_blue_diamond: [:package: ios e2e package](null)

status-im-auto commented:
#### :heavy_check_mark: [status-react/prs/windows/PR-7032#4](https://ci.status.im/job/status-react/job/prs/job/windows/job/PR-7032/4/) :small_blue_diamond: ~17 min :small_blue_diamond: a46ed2cc :small_blue_diamond: [:package: windows package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-181213-082308-a46ed2-pr.exe)

status-im-auto commented:
#### :heavy_check_mark: [status-react/prs/ios/PR-7032#4](https://ci.status.im/job/status-react/job/prs/job/ios/job/PR-7032/4/) :small_blue_diamond: ~17 min :small_blue_diamond: a46ed2cc :small_blue_diamond: [:package: ios package](https://i.diawi.com/M6SJSU)

status-im-auto commented:
#### :heavy_check_mark: [status-react/prs/android/PR-7032#4](https://ci.status.im/job/status-react/job/prs/job/android/job/PR-7032/4/) :small_blue_diamond: ~19 min :small_blue_diamond: a46ed2cc :small_blue_diamond: [:package: android package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-181213-082304-a46ed2-pr.apk)

status-im-auto commented:
#### :heavy_check_mark: [status-react/prs/linux/PR-7032#4](https://ci.status.im/job/status-react/job/prs/job/linux/job/PR-7032/4/) :small_blue_diamond: ~21 min :small_blue_diamond: a46ed2cc :small_blue_diamond: [:package: linux package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-181213-082307-a46ed2-pr.AppImage)

status-im-auto commented:
#### :heavy_check_mark: [status-react/prs/android/PR-7032#5](https://ci.status.im/job/status-react/job/prs/job/android/job/PR-7032/5/) :small_blue_diamond: ~13 min :small_blue_diamond: 6f704a18 :small_blue_diamond: [:package: android package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-181213-094724-6f704a-pr.apk)

status-im-auto commented:
#### :heavy_check_mark: [status-react/prs/macos/PR-7032#5](https://ci.status.im/job/status-react/job/prs/job/macos/job/PR-7032/5/) :small_blue_diamond: ~14 min :small_blue_diamond: 6f704a18 :small_blue_diamond: [:package: macos package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-181213-094727-6f704a-pr.dmg)

status-im-auto commented:
#### :heavy_check_mark: [status-react/prs/ios-e2e/PR-7032#5](https://ci.status.im/job/status-react/job/prs/job/ios-e2e/job/PR-7032/5/) :small_blue_diamond: ~15 min :small_blue_diamond: 6f704a18 :small_blue_diamond: [:package: ios e2e package](null)

status-im-auto commented:
#### :heavy_check_mark: [status-react/prs/ios/PR-7032#5](https://ci.status.im/job/status-react/job/prs/job/ios/job/PR-7032/5/) :small_blue_diamond: ~16 min :small_blue_diamond: 6f704a18 :small_blue_diamond: [:package: ios package](https://i.diawi.com/hJvhrj)

status-im-auto commented:
#### :heavy_check_mark: [status-react/prs/windows/PR-7032#5](https://ci.status.im/job/status-react/job/prs/job/windows/job/PR-7032/5/) :small_blue_diamond: ~16 min :small_blue_diamond: 6f704a18 :small_blue_diamond: [:package: windows package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-181213-094728-6f704a-pr.exe)

status-im-auto commented:
#### :heavy_check_mark: [status-react/prs/android-e2e/PR-7032#5](https://ci.status.im/job/status-react/job/prs/job/android-e2e/job/PR-7032/5/) :small_blue_diamond: ~17 min :small_blue_diamond: 6f704a18 :small_blue_diamond: [:package: android e2e package](https://status-im.ams3.digitaloceanspaces.com/StatusIm-181213-094722-6f704a-e2e.apk)

status-im-auto commented:
#### :heavy_check_mark: [status-react/prs/linux/PR-7032#5](https://ci.status.im/job/status-react/job/prs/job/linux/job/PR-7032/5/) :small_blue_diamond: ~20 min :small_blue_diamond: 6f704a18 :small_blue_diamond: [:package: linux package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-181213-094726-6f704a-pr.AppImage)

asemiankevich commented:
@rasom `null is not an object (evaluating 'e.objects')` shown when creating account

![image](https://user-images.githubusercontent.com/28752642/49991760-258be280-ff93-11e8-90eb-2397e5112114.png)

rasom commented:
@asemiankevich should be fixed now, rebase mess

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 92841f0a | [#6](https://ci.status.im/job/status-react/job/prs/job/macos/job/PR-7032/6/) | 7:42:19 PM | ~12 min | `macos` | [:package: package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-181214-192940-92841f-pr.dmg) |
| :heavy_check_mark: | 92841f0a | [#6](https://ci.status.im/job/status-react/job/prs/job/android/job/PR-7032/6/) | 7:44:26 PM | ~14 min | `android` | [:package: package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-181214-192936-92841f-pr.apk) |
| :heavy_check_mark: | 92841f0a | [#6](https://ci.status.im/job/status-react/job/prs/job/android-e2e/job/PR-7032/6/) | 7:44:36 PM | ~15 min | `android-e2e` | [:package: package](https://status-im.ams3.digitaloceanspaces.com/StatusIm-181214-192935-92841f-e2e.apk) |
| :heavy_check_mark: | 92841f0a | [#6](https://ci.status.im/job/status-react/job/prs/job/ios/job/PR-7032/6/) | 7:46:45 PM | ~17 min | `ios` | [:package: package](https://i.diawi.com/LM8Zov) |
| :heavy_check_mark: | 92841f0a | [#6](https://ci.status.im/job/status-react/job/prs/job/windows/job/PR-7032/6/) | 7:47:47 PM | ~18 min | `windows` | [:package: package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-181214-192941-92841f-pr.exe) |
| :heavy_check_mark: | 92841f0a | [#6](https://ci.status.im/job/status-react/job/prs/job/linux/job/PR-7032/6/) | 7:50:27 PM | ~20 min | `linux` | [:package: package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-181214-192939-92841f-pr.AppImage) |
| :heavy_check_mark: | 92841f0a | [#6](https://ci.status.im/job/status-react/job/prs/job/ios-e2e/job/PR-7032/6/) | 7:51:15 PM | ~21 min | `ios-e2e` | [:package: package](https://status-im.ams3.digitaloceanspaces.com/StatusIm-181214-192937-92841f-e2e.app.zip) |
| | | | | | | |
| :heavy_check_mark: | 7ff35b9f | [#7](https://ci.status.im/job/status-react/job/prs/job/android-e2e/job/PR-7032/7/) | 11:40:14 AM | ~14 min | `android-e2e` | [:package: package](https://status-im.ams3.digitaloceanspaces.com/StatusIm-181217-112526-7ff35b-e2e.apk) |
| :heavy_check_mark: | 7ff35b9f | [#7](https://ci.status.im/job/status-react/job/prs/job/ios-e2e/job/PR-7032/7/) | 11:40:15 AM | ~14 min | `ios-e2e` | [:package: package](https://status-im.ams3.digitaloceanspaces.com/StatusIm-181217-112529-7ff35b-e2e.app.zip) |
| :heavy_check_mark: | 7ff35b9f | [#7](https://ci.status.im/job/status-react/job/prs/job/android/job/PR-7032/7/) | 11:41:48 AM | ~16 min | `android` | [:package: package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-181217-112528-7ff35b-pr.apk) |
| :heavy_check_mark: | 7ff35b9f | [#7](https://ci.status.im/job/status-react/job/prs/job/windows/job/PR-7032/7/) | 11:43:05 AM | ~17 min | `windows` | [:package: package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-181217-112532-7ff35b-pr.exe) |
| :heavy_check_mark: | 7ff35b9f | [#7](https://ci.status.im/job/status-react/job/prs/job/linux/job/PR-7032/7/) | 11:44:31 AM | ~19 min | `linux` | [:package: package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-181217-112530-7ff35b-pr.AppImage) |

asemiankevich commented:
@rasom `Log out` button is disabled for a while when changing accounts ( i have 1 new and 1 recovered) , so every time i am switching between them - this button seems to be disabled for several seconds (2-3 seconds visually). Is it expected?

![image](https://user-images.githubusercontent.com/28752642/50075059-e2c64680-01ee-11e9-8181-459ac1889158.png)

rasom commented:
yes @asemiankevich, it is expected, atm the best we can do

asemiankevich commented:
> yes @asemiankevich, it is expected, atm the best we can do

okay thanks!
